docs/design/vm-metadata-04-sblk.md

```
@@ -0,0 +1,1353 @@
+  如果 payload 超出当前 Slot 容量：
+    new_lv_size = calculate_extend_size(current_lv_size, required)
+    执行 lvextend
+    new_layout = calculate_slot_layout(new_lv_size)
+  否则：
+    new_layout = 当前 Header 中的布局（offset + capacity 不变）
+```
+
+#### 前置步骤 — 确定 op_type（控制面指定）
+
+> **op_type 由控制面指定**：`@MetadataImpact(CONFIG)` → `op_type = CONFIG_UPDATE (1)`，`@MetadataImpact(STORAGE)` → `op_type = STORAGE_CHANGE (2)`。管理层面通过 `storageStructureChange` 字段贯穿整条消息链（`UpdateVmInstanceMetadataMsg` → `WriteVmMetadataToPrimaryStorageMsg` → Agent command），Agent 收到命令时直接使用该值，无需自行读取旧 payload 做 diff。
+
```

markdownlint-cli2 (0.21.0)

[warning] Blank line inside blockquote

(MD028, no-blanks-blockquote)